packages/sdk/src/utils/index.ts

```diff
@@ -0,0 +1,320 @@
+import * as ecc from "@bitcoinerlab/secp256k1";
+import { BIP32Interface } from "bip32";
+import {
+  Network as BitcoinNetwork,
+  Payment,
+  PaymentCreator,
+  Psbt,
+  Signer,
+  Transaction,
+  crypto,
+  initEccLib,
+  networks,
+  payments,
+} from "bitcoinjs-lib";
+import { Buffer } from "buffer";
+import ECPairFactory from "ecpair";
+import { ADDRESS_TYPE_TO_FORMAT } from "../addresses/constants";
+import { OrditSDKError } from "../errors";
+import type { AddressFormat, AddressType } from "../addresses/types";
+import type { Network } from "../config/types";
+import type { UTXO } from "../transactions/types";
+import type {
+  BufferOrHex,
+  EncodeDecodeObjectOptions,
+  GetScriptTypeResponse,
+  IsBitcoinPaymentResponse,
+  NestedObject,
+  OneOfAllDataFormats,
+} from "./types";
+
+export function getNetwork(value: Network) {
+  if (value === "mainnet") {
+    return networks.bitcoin;
+  }
+
+  return networks[value];
+}
+
+export function createTransaction(
+  key: Buffer,
+  type: AddressType,
+  network: Network | BitcoinNetwork,
+  paymentOptions?: Payment,
+) {
+  initEccLib(ecc);
+  const networkObj =
+    typeof network === "string" ? getNetwork(network) : network;
+
+  if (type === "p2tr") {
+    return payments.p2tr({
+      internalPubkey: key,
+      network: networkObj,
+      ...paymentOptions,
+    });
+  }
+
+  if (type === "p2sh") {
+    return payments.p2sh({
+      redeem: payments.p2wpkh({ pubkey: key, network: networkObj }),
+      network: networkObj,
+    });
+  }
+
+  return payments[type]({ pubkey: key, network: networkObj });
+}
+
+export function getDerivationPath(
+  formatType: AddressFormat,
+  account = 0,
+  addressIndex = 0,
+) {
+  const pathFormat: Record<AddressFormat, string> = {
+    legacy: `m/44'/0'/${account}'/0/${addressIndex}`,
+    "p2sh-p2wpkh": `m/49'/0'/${account}'/0/${addressIndex}`,
+    segwit: `m/84'/0'/${account}'/0/${addressIndex}`,
+    taproot: `m/86'/0'/${account}'/0/${addressIndex}`,
+  };
+  return pathFormat[formatType];
+}
+
+export function hdNodeToChild(
+  node: BIP32Interface,
+  formatType: AddressFormat = "legacy",
+  addressIndex = 0,
+  account = 0,
+) {
+  const fullDerivationPath = getDerivationPath(
+    formatType,
+    account,
+    addressIndex,
+  );
+
+  return node.derivePath(fullDerivationPath);
+}
+
+export function toXOnly(pubkey: Buffer): Buffer {
+  return pubkey.subarray(1, 33);
+}
+
+export function tweakSigner(
+  signer: Signer,
+  opts: { tweakHash?: Buffer; network?: BitcoinNetwork } = {},
+): Signer {
+  const ECPair = ECPairFactory(ecc);
+
+  // eslint-disable-next-line @typescript-eslint/ban-ts-comment
+  // @ts-ignore
```

Comment:
@iamcrazycoder why ts-ignore?

Reply:
Looking at this, I can only say that `privateKey` hasn't been exposed in the type interface. 

fwiw, the code is just accessing private key and not injecting it.

Reply:
I've fixed this type and removed ts-ignore.